src/js/button.js

```diff
@@ -7,13 +12,13 @@
  * @class
  * @constructor
  */
-vjs.Button = vjs.Component.extend({
+var Button = Component.extend({
   /**
    * @constructor
    * @inheritDoc
    */
   init: function(player, options){
```

Comment:
Actually, with the plugin, it hopefully won't require action from plugin authors but rather vjs and plugin users. "Upgrade to 5.0, add migration plugin". We have a small amount of plugins right now that we could probably go through each one once 5.0 is in alpha or something with a migration plugin and open up issues on the progres.